tests/test_candle_data.py

```diff
@@ -34,7 +34,7 @@ def test_generate_candle_data(synthetic_candles):
     assert universe.get_pair_count() == 1
     assert universe.get_candle_count() == 4
 
```

Comment:
Can you add an integration test.

Test cases needed

- Read fresh data
- Read cached data
- Purge cache
- Fetch prices
- Fetch lending rates
- Use a specific folder under `/tmp` folder for tests.